cmd/fake/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var counts, errorRate, errors int
+var errorCodes []int
+
+func replyOK(w http.ResponseWriter, r *http.Request) {
+	message := strings.TrimPrefix("/", r.URL.Path)
+	message = message + " served ok"
+
+	responseCode := http.StatusOK
+	w.WriteHeader(responseCode)
+	w.Write([]byte(message))
+}
+
+func replyError(w http.ResponseWriter, r *http.Request) {
+	message := strings.TrimPrefix("/", r.URL.Path)
+	message = "Errored request to " + message
+
+	responseCode := errorCodes[rand.Intn(len(errorCodes))]
+	w.WriteHeader(responseCode)
+	w.Write([]byte(message))
+}
+
+func countRequests(w http.ResponseWriter, r *http.Request) {
+	counts = counts + 1
+	if errorRate > 0 && counts%(100/errorRate) == 0 {
```

Comment:
I'm unclear on the logic here. We provide the rate of errors as
percent, and then you divide 100 per the rate... Can you clarify the
logic in a comment, please?